tests/ui/type-alias-impl-trait/rpit_tait_equality_in_canonical_query.rs

```diff
@@ -0,0 +1,23 @@
+//! This tries to prove the APIT's bounds in a canonical query,
+//! which doesn't know anything about the defining scope of either
+//! opaque type and thus makes a random choice as to which opaque type
+//! becomes the hidden type of the other. When we leave the canonical
+//! query, we attempt to actually check the defining anchor, but now we
+//! have a situation where the RPIT gets constrained outside its anchor.
+
+// check-pass
```

Comment:
Given that this works in the new solver already:

```suggestion
// revisions: current next
//[next] compile-flags: -Ztrait-solver=next
// check-pass
```